src/DataFramesMeta.jl

```diff
@@ -868,4 +922,82 @@ macro select(x, args...)
     esc(select_helper(x, args...))
 end
 
+
+##############################################################################
+##
+## @select! - in-place select and transform columns
+##
+##############################################################################
+
+function select!_helper(x, args...)
+    t = (fun_to_vec(arg) for arg in args)
+
+    quote
+        $DataFrames.select!($x, $(t...))
+    end
+end
+
+"""
+    @select!(d, e...)
+
+Mutate `d` in-place to retain only columns or transformations specified by `e` and return it.
+
+### Arguments
+
+* `d` : an AbstractDataFrame
+* `e` :  keyword arguments specifying new columns in terms of existing columns
+  or symbols to specify existing columns
+
+### Returns
+
+* `::AbstractDataFrame`
```

Comment:
```suggestion
* `::DataFrame`
```

Reply:
I don't quite understand the subtlety of `AbstractDataFrame` and `DataFrame` as I only deal with the standard `DataFrame`. 
My naive understanding is since we are calling `select` directly, if they could return generically so does this package?

If not I guess we also need to change many other occurrence of `AbstractDataFrame` as well. Or this only applies to mutating operations?

Reply:
There are only 2 types of `AbstractDataFrame` currently, a `DataFrame` and a `SubDataFrame`. 

`@select`, `@transform`, and `@combine` all return `DataFrame`s. If we are saying they return `AbstractDataFrame`s then that's not ideal. 

You can change it, but I can also fix it in a small PR after this one, so it's no big deal.